cargo-fix/tests/all/smoke.rs

```diff
@@ -87,3 +87,44 @@ fn tricky_ampersand() {
         .stderr(stderr)
         .run();
 }
+
+#[test]
+fn preserve_line_endings() {
+    let p = project()
+        .file("src/lib.rs", "\
+            fn add(a: &u32) -> u32 { a + 1 }\r\n\
+            pub fn foo() -> u32 { add(1) }\r\n\
+        ")
+        .build();
+
+    p.expect_cmd("cargo fix").run();
+    assert!(p.read("src/lib.rs").contains("\r\n"));
+}
+
+#[test]
+fn multiple_suggestions_for_the_same_thing() {
+    let p = project()
+        .file("src/lib.rs", "\
+            fn main() {
+                let xs = vec![String::from(\"foo\")];
+                // error: no diplay in scope, and there are two options
+                // (std::path::Display and std::fmt::Display)
+                let d: &Display = &xs;
+                println!(\"{}\", d);
+            }
+        ")
+        .build();
+
+    let stderr = "\
+[CHECKING] foo v0.1.0 (CWD)
+error: Cannot replace slice of data that was already replaced
```

Comment:
Hm I'm not sure that this is the UI we're gonna want long-term, I think we probably want to just skip auto-fixing code and let warnings like these naturally bubble up to the user?

Could we add some logic to avoid attempting to apply suggestions that look like they're gonna fail? (or otherwise just keep track of suggestions that fail to apply)

Reply:
Yes that's why I was asking for a better test case :) I opened #78 for the intended user experience

Reply:
Hm could we tweak this test to assert that the warning isn't updated here? I think it's probably most conservative to start out by not applying any suggestions when they're ambiguous about what to do. Anything we don't fix will end up being seen by the end user anyway

Reply:
Yeah. Do you want me to do this here or in another PR? (The behavior isn't changed here AFAIK)

Reply:
Ah ok let's do that in a follow-up then